pkg/patcher/scheduler_patcher_openshift.go

```diff
@@ -263,13 +264,17 @@ func createOpenshiftConfigMapObject(config string, useOpenshiftSecondarySchedule
 	}
 }
 
-func (p *SchedulerPatcher) patchSecondaryScheduler(ctx context.Context) (*ssv1.SecondaryScheduler, error) {
+func (p *SchedulerPatcher) patchSecondaryScheduler(ctx context.Context, csi *csibaremetalv1.Deployment) (*ssv1.SecondaryScheduler, error) {
 	secondaryScheduler := &ssv1.SecondaryScheduler{}
+	csiOpenshiftSecondarySchedulerImage := common.ConstructFullImageName(
+		csi.Spec.Scheduler.OpenshiftSecondaryScheduler.Image, csi.Spec.GlobalRegistry)
```

Comment:
what if user deploy using an old charts and with new image tag ?   in such case,  the scheduler image, tag will be null, should we handle this ?

Reply:
> what if user deploy using an old charts and with new image tag ? in such case, the scheduler image, tag will be null, should we handle this ?

In the CSI standard installation, the same CSI_OPERATOR_VERSION env var has been used for both csi-charts and csi baremetal operator and we hardly encounter the case that use an old charts with new image tag.
But, as you pointed out, I will handle this corner case to ensure CSI installation can work well in all cases.

Reply:
@libzhang 
I've added the logic to handle the case of using old charts and new operator image and also the case of using new charts and new operator image but set secondary scheduler image as null